### PR TITLE
[renault] Add channel for pausing/resuming charging

### PR DESCRIPTION
**Pull request by fifipil909**
Adding Pause / Resume API endpoint.

On some Vehicule type, changing the Charge mode to pause / Resume the charge is not working.
Another endpoint is required.

https://renault-api.readthedocs.io/en/latest/endpoints/vehicle_kcm_actions.charge-pause-resume.html

Tested on 3.4 and 4.0.0

Vincent.

### CODE REVIEW

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the PR. Please see my proposal about modeling this pause/resume mode into a channel.

File: bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml
Comment (fifipil909) [reply bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml:153]: I follow the same pattern as the other Channel (like Hvac) but I agree with you a switch might be better here

**Comment by dougculnane** [comment]
@fifipil909 you have to sign off your commits.

**Review by dougculnane — CHANGES_REQUESTED** [review]
Thanks Vincent this looks like it fits with the existing code well keeping the code consistent. So it will get approval from me but you have to sign off your commits and the feedback from @jlaur is very good and fits with feedback I got about the PRs I have submitted.

One thing I would like to see is your valuable comment about the chargingmode etc.. added to the README https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.renault#limitations  My hope is that making it clear that not all things work on all cars will reduce the user frustration.  I use the chargingmode for the example battery limit rule, so maybe a comment here is approximate too and will help some users.

**Comment by fifipil909** [comment]
> @fifipil909 you have to sign off your commits.

Done. 
But I don't know why Something from Homekit was include in my commit. 
I never touch this file !

**Comment by dougculnane** [comment]
Hi Vincent,  This can get very confusing but adding the  git --signoff to your commits and rebasing can touch other commits.  It looks like the homebase changes from somewhere got added in here.  What we need it your --signoff commits in this branch with the changes suggested.  If you need help with this I can try to help next week.

**Comment by fifipil909** [comment]
Sorry for the confusion. I'm not using rebase a lot.

I rebase correctly and remove the other commits.
No the PR do have only 1 commit and it's signed.

Hope is fine.

**Comment by dougculnane** [comment]
@fifipil909 Looks good and the checks have passed. Can you go the changes requested and i think we are there.  I will try to find time to test this to see if the pause charge feature works on the Zoe.

File: bundles/org.openhab.binding.renault/src/main/resources/OH-INF/i18n/renault_fr.properties
Comment (fifipil909) [reply bundles/org.openhab.binding.renault/src/main/resources/OH-INF/i18n/renault_fr.properties:65]: Hi @jlaur 
Do you mean that I need to pull the English word in [renault_fr.properties] for all of these channel options, and they will be automatically translated ?

Thanks

Comment (jlaur) [reply bundles/org.openhab.binding.renault/src/main/resources/OH-INF/i18n/renault_fr.properties:65]: No, you need to revert the changes in `renault_fr.properties`. After the PR has been merged, the translations can be submitted in Crowdin, and a PR will automatically be created so the texts can finally be merged into the codebase.

File: bundles/org.openhab.binding.renault/README.md
Comment (jlaur) [inline bundles/org.openhab.binding.renault/README.md]: Now a `Switch` item.

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/Car.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/Car.java:165]: Will this work? `PauseMode` has been redefined to contain members `ON` and `OFF`, where it was previously `PAUSE` and `RESUME`. I assume the API still returns `PAUSE` and `RESUME`?

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java:220]: Channel is now of type `Switch`.

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java:223]: Probably you only want to do this for `InterruptedException`.

File: bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml:154]: Description missing.

File: bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml:19]: Consider if `paused` would be a better name for a `Switch` item.

File: bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/resources/OH-INF/thing/thing-types.xml:156]:
```suggestion
		<state readOnly="false"></state>
```

**Review by jlaur — CHANGES_REQUESTED** [review]
A few new comments after latest changes. Please have a look at the DCO also, it seems the new commits were not properly signed.

File: bundles/org.openhab.binding.renault/README.md
Comment (jlaur) [inline bundles/org.openhab.binding.renault/README.md:40]:
```suggestion
| pausemode              | Switch             | Pause mode.                                     | No        |
```

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java
Comment (fifipil909) [reply bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java:224]: I make a more clear exception handle on my very last commit
Not sure this is what you would like.

File: bundles/org.openhab.binding.renault/README.md
Comment (jlaur) [inline bundles/org.openhab.binding.renault/README.md:63]:
```suggestion
The `pause` and `chargingmode` may not work on some cars.
As an example, `chargingmode` does not work on Dacia Spring cars.
```

File: bundles/org.openhab.binding.renault/README.md
Comment (jlaur) [inline bundles/org.openhab.binding.renault/README.md:83]:
```suggestion
        Default icon="switch" item=RenaultCar_Pause
```

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/Car.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/Car.java:49]: It seems that this class in general abstracts openHAB types, so probably this should be:
```suggestion
    private boolean pausemode = false;
```

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/Car.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/Car.java:303]:
```suggestion
    public boolean getPauseMode() {
```

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/Car.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/Car.java:355]:
```suggestion
    public void setPauseMode(boolean pausemode) {
```

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/MyRenaultHttpSession.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/MyRenaultHttpSession.java:278]:
```suggestion
    public void actionPause(boolean pause)
```

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/MyRenaultHttpSession.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/api/MyRenaultHttpSession.java:281]:
```suggestion
        final String apiMode = pause ? "pause" : "resume";
```

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java:219]:
```suggestion
                            boolean pause = OnOffType.ON == command;
                            httpSession.initSesssion(car);
                            httpSession.actionPause(pause);
                            car.setPauseMode(pause);
                            updateState(CHANNEL_PAUSE_MODE, command);
```

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java
Comment (jlaur) [reply bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/handler/RenaultHandler.java:224]: You need a separate catch for `InterruptedException`. See how this is handled in the code above.

File: bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/RenaultBindingConstants.java
Comment (jlaur) [inline bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/RenaultBindingConstants.java:37]:
```suggestion
    public static final String CHANNEL_PAUSE = "pause";
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Next round. I provided a few very specific suggestions to avoid misunderstandings. These are related to the comment about not using openHAB types in the Car API classes.

**Review by dougculnane — APPROVED** [review]
LGTM.  Thanks Vincent.

**Review by jlaur — APPROVED** [review]
LGTM

File: bundles/org.openhab.binding.renault/README.md
Comment (jlaur) [inline bundles/org.openhab.binding.renault/README.md:64]:
```suggestion
As an example, 'chargingmode' does not work on Dacia Spring cars.
```

**Comment by lolodomo** [comment]
Was the new thing upgrade mechanism implemented?

**Comment by jlaur** [comment]
> Was the new thing upgrade mechanism implemented?

No, I need to get used to this (and we need some documentation). Created #14572.

**Pull request merged**
